pkg/configuration/environment.go

```diff
@@ -0,0 +1,122 @@
+package configuration
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/subosito/gotenv"
+)
+
+// LoadConfiguredEnvironment updates the environment with local configuration. Precedence as follows:
+//  1. std folder-based config files
+//  2. given command-line parameter config file
+//  3. std config file in home directory
+//  4. global shell configuration
+func LoadConfiguredEnvironment(config Configuration) {
```

Comment:
Suggestion: while it seems reasonable that this is part of the `configuration` package, I would like to not export any of these functions and load the environment inside the configuration automatically. The reason for this is, that I would like to continue what viper does and hide implementation details about the source of configuration values. In the future we might even have an improved design to add configuration provider besides what is existing in viper. but since this is not available yet, let's try to remove this from the public API of the package.

Having said this, we need a way to enable additional sources of configuration, for example as a configuration value. This should also ensure backwards compatibility for the CLI where we don't want to create regression by reading from an additional source.

Reply:
I need to call it from LS before a scan to get folder specific environment settings. As the legacycli workflow would need to do, too. So we'll need it as it needs to be at runtime. Or we need to integrate it into being called each time when a workflow is invoked. 

Regarding the consistent configuration, I'd like that - should we set each loaded variable into the config manually, or how would you propose to handle that?